### ISSUE TITLE
 failed calling webhook "webhook.cert-manager.io"

### ISSUE THREAD

**Issue by toddlerds** [kubeflow/kubeflow#5230]
https://www.kubeflow.org/docs/started/k8s/kfctl-k8s-istio/

kfctl apply -V -f ${CONFIG_URI}

error:
Encountered error applying application cert-manager:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout884432936": Internal error occurred: failed calling webhook "webhook.cert-manager.io": the server is currently unable to handle the request  filename="kustomize/kustomize.go:266"

**Comment by MartinForReal** [comment]
is cert-manager unavailable for processing requests?

**Comment by yangjinlogic** [comment]
I met the same problem

**Comment by DavidSpek** [comment]
What version of Kubernetes are you all running? Was cert-manager already deployed on the cluster before deploying Kubeflow? If cert-manager is already installed on the cluster before you deploy Kubeflow (and you do not remove cert-manager from the installation) you can face this issue.

**Comment by jednymslowem** [comment]
I have the same issue. The output is now a bit more verbose:
```
WARN[0313] Encountered error applying application kubeflow-apps:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout038522058": CustomResourceDefinition.apiextensions.k8s.io "seldondeployments.machinelearning.seldon.io" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[explainer].properties[containerSpec].properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]  filename="kustomize/kustomize.go:266"
```
I'm using kfctl_k8s_istio.v1.1.0.yaml and Kubernetes version 1.18.8 (nothing was deployed to the cluster before).

**Comment by DavidSpek** [comment]
@jednymslowem The issue you are having is not related to the webhook.cert-manager.io but rather seldon. Please see the following comment I made which links to other issues that are the same and a workaround for the time being. https://github.com/kubeflow/website/issues/2206#issuecomment-698234904

**Comment by jbottum** [comment]
Hi @toddlerds did you get this resolved?  which platform (AWS, GCP, On-prem, are you running on) ?

**Comment by KakarotSonGokuShenron** [comment]
> What version of Kubernetes are you all running? Was cert-manager already deployed on the cluster before deploying Kubeflow? If cert-manager is already installed on the cluster before you deploy Kubeflow (and you do not remove cert-manager from the installation) you can face this issue. Is there a pod that is going into a CrashLoopBackOff? If so, what are the logs from that pod?

So @DavidSpek, what if you are trying to deploy to an existing cluster that already has cert-manager installed? I'm facing issues with the existing cert-manager being wiped out by the one from kubeflow and wondering how to deploy kubeflow while keeping the cluster's existing cert-manager.

**Comment by DavidSpek** [comment]
@KakarotSonGokuShenron You can remove cert-manager from the kfctl deployment yaml. Similar to what you do when using an existing istio installation. https://www.kubeflow.org/docs/started/k8s/kfctl-istio-dex/#notes-on-the-configuration-file

**Comment by TT310** [comment]
ERRO[0634] Permanently failed applying application cert-manager:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout781015294": Internal error occurred: failed calling webhook "webhook.cert-manager.io": the server is currently unable to handle the request  filename="kustomize/kustomize.go:288"
Error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout781015294": Internal error occurred: failed calling webhook "webhook.cert-manager.io": the server is currently unable to handle the request

how can I to do?

**Comment by richardagalvez** [comment]
For everyone encountering this problem, wanted to share what I found as this thread never fully resolved a common issue I've encountered. 

It turns out the `cert-manager` application could take a bit of time to deploy, and for some reason, not succeed at deploying at all. For me in particular, the pods it deploys stay in ContainerCreating status for a while and eventually error out with this error (after running `kubectl describe pod <cert-manager troubled pod name>`):

```
Events:
  Type     Reason      Age                    From     Message
  ----     ------      ----                   ----     -------
  Warning  FailedSync  3m28s (x330 over 96m)  kubelet  error determining status: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

My work around for this was to delete the pod directly (with a `kubectl delete pod <cert-manager troubled pod name> -n cert-manager`). This allows the cert-manager deployment to try to deploy another pod, which works for me on the second time. After I check that the pod is properly running, I re-run the kfctl install yaml command to be sure I didn't miss any parts of the kubeflow install. In most cases, you'll see all of the resources will just log that they are unchanged. Hopefully this solves your issue.
